Incompatible pointer types in imageioJPEG.c

Build fails on GCC 14 with multiple instances of:

~~~
/home/andrew/builder/icedtea8/openjdk-boot/jdk/src/share/native/sun/awt/image/jpeg/imageioJPEG.c: In function 'Java_com_sun_imageio_plugins_jpeg_JPEGImageWriter_writeImage':
/home/andrew/builder/icedtea8/openjdk-boot/jdk/src/share/native/sun/awt/image/jpeg/imageioJPEG.c:2857:19: error: passing argument 1 of 'freeArray' from incompatible pointer type [-Wincompatible-pointer-types]
 2857 |         freeArray(scale, numBands);
      |                   ^~~~~
      |                   |
      |                   UINT8 ** {aka unsigned char **}
/home/andrew/builder/icedtea8/openjdk-boot/jdk/src/share/native/sun/awt/image/jpeg/imageioJPEG.c:2693:30: note: expected 'void **' but argument is of type 'UINT8 **' {aka 'unsigned char **'}
 2693 | static void freeArray(void** arr, jint size) {
      |                       ~~~~~~~^~~
~~~

Seems to be the same as [JDK-8162545](https://bugs.openjdk.org/browse/JDK-8162545) which has a one-line fix.  `freeArray` is only used with `scale` which is defined as `UINT8** scale = NULL;`

--- a/jdk/src/share/native/sun/awt/image/jpeg/imageioJPEG.c
+++ b/jdk/src/share/native/sun/awt/image/jpeg/imageioJPEG.c
@@ -2690,7 +2690,7 @@ Java_com_sun_imageio_plugins_jpeg_JPEGImageWriter_writeTables
     RELEASE_ARRAYS(env, data, NULL);
 }
 
-static void freeArray(void** arr, jint size) {
+static void freeArray(UINT8** arr, jint size) {
     int i;
     if (arr != NULL) {
         for (i = 0; i < size; i++) {

